tools: scripts: Added Makefile support

scuciurean opened the pull request:
Added Makefile support for the Xilinx platform.

dbogdan commented:
Is this ready to be reviewed? :)

scuciurean commented:
Changelog -> V2:

1. Added missing _app_config.h_ to src.mk files

scuciurean commented:
> Is this ready to be reviewed? :)



> Is this ready to be reviewed? :)

Certainly not by commodo

commodo commented on tools/scripts/Makefile:
```diff
@@ -0,0 +1,265 @@
+export PLATFORM
+
+NOOS_DIR 	  = $(CURDIR)/../..
+ROOT_DIR 	  = ../..
```
doer ROOT_DIR get used?
if yes, is ROOT_DIR == NOOS_DIR?

scuciurean replied:
Removed ROOT_DIR. Thanks

commodo commented on tools/scripts/Makefile:
```diff
@@ -0,0 +1,265 @@
+	 -MMD 								\
+	 -MP								\
+	 -lm						
+	#-Werror
```
this looks commented;
maybe drop?

scuciurean replied:
That flag was left commented on purpose. Will get removed after all the project/drivers warnings are removed.

scuciurean commented:
Changelog -> V3:

1. Removed unused Makefile variable

commodo commented on tools/scripts/Makefile:
```diff
@@ -0,0 +1,265 @@
+# Zynq
+ifeq (ps7_cortexa9_0,$(strip $(PROCESSOR_CORE)))
+
+CC = arm-none-eabi-gcc
```
it's a good idea to do `CC := arm-none-eabi-gcc` here; to make sure that it gets overriden;
there could be a chance that `CC` has `gcc` assigned;

commodo commented on tools/scripts/Makefile:
```diff
@@ -0,0 +1,265 @@
+%.o : %.c
+	$(ECHO) [CC]	$(notdir $<) '\n'
+	$(MUTE)$(CC) $(CFLAGS) $(INC_PATHS) $(EXTRA_INCS)		\
+	-c $(addprefix $(BUILD_DIR)/app/src/,$(notdir $<))		\
```
curios: why `$(addprefix)` ?
this looks a bit complicated;

usually, it's a good idea for OBJS to be defined with full paths

scuciurean replied:
While building the project, one of the steps is to copy the srcs from the no-Os repo to a local srcs folder. With `$(addprefix)` and `$(notdir)` I redefine [the path of] the compiled sources to be the local ones, and not the repository's. Also, while generating the object files, I prefer to save them in a separate folder called _obj_.

commodo commented on tools/scripts/Makefile:
```diff
@@ -0,0 +1,265 @@
+eval-hardware:
+# If the hardware file is not specified, start searching for one
+# Check for .hdf files inside the project directory		
+ifndef $(HARDWARE)
```
why not move everything into a shell script and do `PLATFORM=$(shell script,param1,param2,etc)` ?

scuciurean replied:
Since the makefile offers this functionality, why not use it. Also, in this way, the code is a little bit more organized and not split in several files that will be shared only to this makefile.

commodo commented on tools/scripts/Makefile:
```diff
@@ -0,0 +1,265 @@
+	fi;
+	@$(MAKE) -s compile
+
+xilinx:	xil-pre-cleanup							\
```
why not put all on a single line ?
so: `xilinx: xil-pre-cleanup eval-hardware pre-build`
the way it is now, it is confusing to debug, as deps are indented similarly as rules

commodo commented on tools/scripts/Makefile:
```diff
@@ -0,0 +1,265 @@
+	$(MUTE)rm -rf arch.txt
+# Extract the processor name from the hdf file
+	$(MUTE)xsct $(SCRIPTS_PATH)/read_hdf.tcl $(NULL)		\
+	temp/$(notdir $(HARDWARE))
```
if breaking a new line here, maybe indent the 2nd, to make it more obvious to see they are the same

commodo commented on tools/scripts/Makefile:
```diff
@@ -0,0 +1,265 @@
+	$(MUTE)cp -r $(subst LOCAL_PLATFORM,$(1),$(SRCS))		\
+	$(BUILD_DIR)/app/src/
+	$(MUTE)cp -r $(subst LOCAL_PLATFORM,$(1),$(INCS))		\
+	$(BUILD_DIR)/app/src/
```
if breaking a new line, maybe indent the 2nd to make sure it's the same command;
maybe check this, in all the file

scuciurean commented:
Changelog -> V4:

1. Changed indentation to differentiate the rules form deps.
2. Changed compiler assignation to use the walrus operator (overwrite).

scuciurean commented:
Changelog -> V5:

1. Updated src.mk files with the new path for the axi drivers ( drivers/axi_core )

scuciurean commented:
Changelog -> V6:

1. Removed platform_drivers.h from the target sources in all the projects
